### ISSUE TITLE
X11 uses the Wayland message box code when no video system is initialized

### ISSUE THREAD

**Issue by icculus** [libsdl-org/SDL#5445]
So if you run SDL/test/testmessage.c, you'll notice on X11 you'll get message boxes like this...

![2022-03-24_14-28](https://user-images.githubusercontent.com/673562/159985961-998aca16-677a-4c62-a405-bf56b536b7af.png)

...until you get to the last one...

![2022-03-24_14-29](https://user-images.githubusercontent.com/673562/159986006-cc7313cf-f9d6-46f6-9fd6-3d2c605d964d.png)

...this is because, without the video subsystem initialized, we use `SDL_MessageboxValidForDriver` to decide what to do, but this returns `SDL_TRUE` immediately if the parent window is NULL.

Which means we try the Wayland code before the X11 code. And the Wayland code just spawns an instance of zenity to show a GTK+ message box.

Problem 1: we should probably choose the X11 driver here, but we definitely want Wayland if this would just end up talking to XWayland, so it's complicated.
Problem 2: SDL_MessageboxValidForDriver needs a better approach, perhaps.
Problem 3: the GTK+ dialog looks _SO MUCH BETTER_. I kinda feel like we should use this on X11 too, and fallback to our own code if zenity isn't available.

**Comment by weirddan455** [comment]
I did some testing on this.  Regarding the problems you stated:

Problem 1: It looks like SDL does choose X11 driver first in the last stable release 2.0.20.  At some point since then code in SDL_ShowMessageBox got changed so it tries Wayland first.

Problem 2: Yea, maybe it should be doing some real detection of whether X11 or Wayland is actually running.

Problem 3:  I made some quick changes to SDL to use Zenity over X11 when there's an SDL window up.  It works but it's not a proper modal dialog like the current X11 code is.  It actually gets parented to my terminal window for some reason.  Could be a problem for full screen programs especially.  If the user clicks anywhere, the error message can get hidden away.
